packages/amplify_flutter/lib/src/utils/equatable.dart

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+import 'package:collection/collection.dart';
+
+/// Provides overrides for [==] and [hashCode] for consistent equality and
+/// hashing.
+mixin AWSEquatable<T extends AWSEquatable<T>> on Object {
```

Comment:
Should this be in core? Right now it is only used in amplify_flutter, but it seems useful elsewhere. I guess we can always move it later so I don't think it is a big deal either way.

Reply:
It will be in `aws_common` eventually. That is where I copied it from. Since I believe that will be published before the end of the year, I just put it here for now.